module2/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,69 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+Compressed compressGrayscale(const Image& compress) {
+    Compressed compressedBitmap;
+
+    int repetitionCounter = 0;
+    for (const auto& row : compress) {
+        uint8_t actualColor = row.at(0);
+        auto lineBegin = row.begin();
+        auto lineEnd = row.end();
+        while (lineBegin < row.begin() + width) {
+            lineEnd = std::find_if(lineBegin, lineEnd,
+                                   [actualColor](uint8_t color) {
+                                       return actualColor != color;
+                                   });
+            repetitionCounter = std::distance(lineBegin, lineEnd);
+            if (repetitionCounter == 0) {
+                repetitionCounter = 1;
+            }
+            compressedBitmap.push_back({actualColor, repetitionCounter});
+            std::advance(lineBegin, repetitionCounter);

```

Comment:
You don't need to do it. `std::find_if` return new begin for you. 
Just use: `lineBegin = lineEnd`

Reply:
also suggest change name from lieEnd to `currentLine` or sth like that.